### DISCUSSION TITLE
Depending another task on 'publishToSonatype' in multi module setup

### DISCUSSION THREAD

**Discussion by skuzzle** [gradle-nexus/publish-plugin#186] [Q&A] [answered]
I'm trying to combine the `publish-plugin` with the `net.researchgate.release` plugin. The latter offers a 'hook' task `afterReleaseBuild` that can be used to execute other tasks in the context of the release process (documented [here](https://github.com/researchgate/gradle-release#custom-release-steps)).

In `build.gradle` of my root project I'm trying to hook in the `publishToSonatype` task like this:
```groovy
plugins {
    id("net.researchgate.release") version "3.0.2"
    id("io.github.gradle-nexus.publish-plugin") version "1.2.0"
}
afterReleaseBuild.dependsOn("publishToSonatype", "...")

release {
    // ...
}

nexusPublishing {
    // ...
}
```

When I run the `afterReleaseBuild` task (either explicitly via `gradlew afterReleaseBuild` or implictly via `gradlew release`) I get the following error:
```
Could not determine the dependencies of task ':afterReleaseBuild'.
> Task with path 'publishToSonatype' not found in root project 'snapshot-tests'.
```

Directly invokig `gradlew publishToSonatype` on the other hand seems to work.

The full project consists of multiple modules, some of which are subject to the sonatype publication (as they contribute files via `maven-publish` plugin), others are not part of the publication. 

I do wonder how to correctly express that the `afterReleaseBuild` task should depend on the successful publication of all elligible child projects.

I have a build scan available here: https://scans.gradle.com/s/53y4kncyxhqpe
The full project setup I'm trying to build can be found here: https://github.com/skuzzle/snapshot-tests/tree/dcec50e7f9021caa2c5cc2fa00b3f1b00fb7a0b6

I'm not quite sure if this question is even related to this plugin or if I'm lacking general understanding of how task dependencies in gradle work.

**Comment by szpak** [comment, marked as answer]
The solution is in a reply to that answer.

---

The commit you mentioned uses Maven. I have found that [`dev`](https://github.com/skuzzle/snapshot-tests/tree/dev) uses Gradle, however, the CI build there seems to be green. Have you been able to resolve your issue in the meantime?

Update. Especially that the build scan shows `The org.gradle.util.ConfigureUtil type has been deprecated.` which should no longer exist with 1.2.0 which is used in the `dev` branch.

Btw, I migrated it from issues to the discussions.

**Reply by skuzzle** [reply]
Uh I confused some stuff here, sorry for that. 
1. I indeed wanted to post a link to the dev branch
2. The build scan I posted was from a build from before I upgraded from `1.1.0` to `1.2.0`. That is why the deprecation was shown there, while the dev branch already pointed to the updated version.

I think I have solved the issue with some tips I found over here :https://github.com/researchgate/gradle-release/issues/138

I've modified the configuration to depend on the `publishToSonatype` task of only those subprojects that are elligible for publication via:

```groovy
tasks.named("afterReleaseBuild").configure {
    dependsOn(provider {
        subprojects
                .findAll { it.getPluginManager().hasPlugin("snapshot-tests.publishing-conventions") }
                .collect { it.getTasks().named("publishToSonatype") }
    })
    dependsOn("commitFilesToGit")
}
```

The CI gods were merciful and finally allowed me to perform a successful release after migrating the whole build to gradle 🎉

Thanks a lot for looking into this and sorry for having wasted your time here. But maybe this discussion helps some folks who are struggeling with the same issue.

**Reply by szpak** [reply]
No problem, thanks for sharing your solution. And "happy gradling" ;).
